src/blocks/kdeconnect.rs

```diff
@@ -84,14 +84,32 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
     let dbus_conn = new_dbus_connection().await?;
     let id = match config.device_id {
         Some(id) => id,
-        None => api.recoverable(|| any_device_id(&dbus_conn)).await?,
+        None => api.recoverable(|| any_device_connected(&dbus_conn)).await?,
     };
 
     let (tx, mut rx) = mpsc::channel(8);
-    let device = Device::new(&dbus_conn, tx, &id).await?;
+    let mut device = Device::new(&dbus_conn, tx, &id).await?;
 
     loop {
-        let connected = device.connected().await;
+        let mut connected = device.connected().await;
+
+        if !connected {
+          let id = match any_device_connected(&dbus_conn).await {
+                Ok(id) => id,
+                Err(_) =>{ 
+                    api.hide().await?;
+                    continue;
+                }
+            };
+
+            let (tx_new, rx_new) = mpsc::channel(8);
+
+            rx = rx_new;
+            
+            device = Device::new(&dbus_conn, tx_new, &id).await?;
```

Comment:
I haven't tested this, but it looks like a task with is spawned in `Device::new()` will panic after `rx = rx_new`, which drops `rx`.